File: Fragments/app/src/main/java/com/example/fragments/MainActivity.java
package com.example.fragments;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.example.fragments.fragment.ContactFragment;
import com.example.fragments.fragment.ConversationFragment;

import java.util.Objects;

public class MainActivity extends AppCompatActivity {

    private Button buttonConversation, buttonContact;
    private ConversationFragment conversationFragment;
    private ContactFragment contactFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Objects.requireNonNull(getSupportActionBar()).setElevation(0);

        buttonConversation = findViewById(R.id.buttonConversation);
        buttonContact = findViewById(R.id.buttonContact);

        conversationFragment = new ConversationFragment();
        contactFragment = new ContactFragment();
        this.initializeFragment(conversationFragment);

        buttonConversation.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                initializeFragment(conversationFragment);
            }
        });

        buttonContact.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                initializeFragment(contactFragment);
            }
        });
    }

    private void initializeFragment(Fragment fragment) {
        // Configurar objeto para fragment
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.frameLayout, fragment);
        fragmentTransaction.commit(); // Encerrar transações
    }
}

File: app/src/main/java/com/example/fragments/MainActivity.kt
package com.example.fragments

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import com.example.fragments.fragments.CallFragment
import com.example.fragments.fragments.ChatFragment

class MainActivity : AppCompatActivity() {

    private lateinit var buttonMarketplace: Button
    private lateinit var buttonCall: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.i("life_cycle", "Activity onCreate")

        initActivity()
        getFragmentChat()
        getFragmentCall()

    }

    override fun onStart() {
        super.onStart()
        Log.i("life_cycle", "Activity onStart")
    }

    override fun onResume() {
        super.onResume()
        Log.i("life_cycle", "Activity onResume")
    }

    override fun onPause() {
        super.onPause()
        Log.i("life_cycle", "Activity onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.i("life_cycle", "Activity onStop")
    }

    override fun onDestroy() {
        Log.i("life_cycle", "Activity onDestroy")
        super.onDestroy()
    }

    private fun initActivity() {
        buttonMarketplace = findViewById(R.id.button_marketplace)
        buttonCall = findViewById(R.id.button_call)
    }

    private fun initFragment(fragment: Fragment) {
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.fragmentContainerView, fragment)
            .commit()
    }

    private fun getFragmentChat() = buttonMarketplace.setOnClickListener {
        val chatFragment = ChatFragment()
        val bundle = bundleOf("category" to "mercado")
        chatFragment.arguments = bundle
        initFragment(chatFragment)
    }
    private fun getFragmentCall() = buttonCall.setOnClickListener { initFragment(CallFragment()) }
}